tests/pytests/unit/states/test_user.py

```diff
@@ -264,3 +264,49 @@ def mock_exists(*args):
             "warndays": 7,
             "inactdays": 0,
         }
+
+
+def test_gecos_field_changes_in_user_present():
+    """
+    Test if the gecos fields change in salt.states.user.present
+    """
+    shadow_info = MagicMock(
+        return_value={"min": 2, "max": 88888, "inact": 77, "warn": 14, "passwd": ""}
+    )
+    shadow_hash = MagicMock(return_value="abcd")
+    mock_info = MagicMock(
+        side_effect=[
+            {
+                "uid": 5000,
+                "gid": 5000,
+                "groups": ["foo"],
+                "home": "/home/foo",
+                "fullname": "Foo Bar",
+                "homephone": "667788",
+            },
+            {
+                "uid": 5000,
+                "gid": 5000,
+                "groups": ["foo"],
+                "home": "/home/foo",
+                "fullname": "Bar Bar",
+                "homephone": "44566",
+            },
+        ]
+    )
+    mock_changes = MagicMock(side_effect=[{"homephone": "667788"}, None])
+    dunder_salt = {
+        "user.info": mock_info,
+        "user.chhomephone": MagicMock(return_value=True),
+        "shadow.info": shadow_info,
+        "shadow.default_hash": shadow_hash,
+        "file.group_to_gid": MagicMock(side_effect=["foo"]),
+        "file.gid_to_group": MagicMock(side_effect=[5000, 5000]),
+    }
+    with patch.dict(user.__grains__, {"kernel": "Linux"}), patch.dict(
+        user.__salt__, dunder_salt
+    ), patch.dict(user.__opts__, {"test": False}), patch.object(
+        user, "_changes", mock_changes
```

Comment:
also FWIW - no need to create a separate mock here. `patch` passes arguments to the underlying Mock, so in this case it could easily be `patch.object(user, "_changes", side_effect=[{"homephone": "667788"}, None])` and it would perform the same.

Makes the test code quite a bit cleaner by eliminating the extra `MagicMock` calls.

Reply:
Understood @wayne, and you're right, that's a much cleaner way.